### ISSUE TITLE
Migrate from Dagger to KOIN

### ISSUE THREAD

**Issue by jumaallan** [jumaallan/android-mpesa-api#46]
Migrate the DI setup to KOIN, from Dagger

**Comment by muth0mi** [comment]
I tried replacing dagger with hilt a while ago and I didn't really find need for a DI library outside the demo app.

**Comment by jumaallan** [comment]
I think for the sample project, we should not use any DI. This will make the example easy to use by anyone who wants to learn how the library works

**Comment by wangerekaharun** [comment]
> I think for the sample project, we should not use any DI. This will make the example easy to use by anyone who wants to learn how the library works

with this in mind you should probably close this issue

**Comment by jumaallan** [comment]
This issue was meant to change the DI inside the library module itself, not the sample app

**Comment by wangerekaharun** [comment]
Seen the library module doesn't have any DI

**Comment by jumaallan** [comment]
Yeah, the re-write was going to use KOIN maybe, or we can just do away with the DI concept on the module

**Comment by jumaallan** [comment]
@andronicus-kim Feel free to contribute to this conversation

**Comment by andronicus-kim** [comment]
I think introducing DI on the module will make it a lot cleaner.

**Comment by jumaallan** [comment]
Great. Let's continue working from your current PR
